main/src/ca/uwaterloo/flix/language/ast/Level.scala

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright 2023 Matthew Lutze
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package ca.uwaterloo.flix.language.ast
+
+import ca.uwaterloo.flix.util.InternalCompilerException
+
+/**
+  * Indicates a nesting depth.
+  *
+  * Each time evaluation enters a region, level increases by one.
+  */
```

Comment:
Currently, we only increase the nesting depth for regions (the scope expression).